client/postit/src/components/SignUp.js

```diff
@@ -1,23 +1,39 @@
 import "bootstrap/dist/css/bootstrap.css";
 import "../css/custom.css";
-import { useState } from "react";
-function SignUp({ onRegistration }) {
+import { useState, useEffect } from "react";
+function SignUp({ onRegistration, onError }) {
   const initialValues = {
     username: "",
     email: "",
     password: "",
   };
   const [values, setValues] = useState(initialValues);
-
+  const [errMsg, setErrMsg] = useState("");
   const handleInputChange = (e) => {
     const { name, value } = e.target;
 
     setValues({ ...values, [name]: value });
   };
 
-  const handleSubmit = (event) => {
+  useEffect(() => {
```

Comment:
This useEffect seems redundant to me, ideally your code says it should run each time any of these change [values.username, values.email, values.password]. Oh I see you are using it as a way to clear your error